### PR TITLE
Feature/my payment

### PR DESCRIPTION
**Pull request by isaacSennerholt**

### CODE REVIEW

File: Hedvig/Screens/ProfileScreen/Screens/MyPayment/MyPayment.swift [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+extension MyPayment: Presentable {
+    func materialize() -> (UIViewController, Disposable) {
+        let bag = DisposeBag()
+
+        let viewController = UIViewController()
+        viewController.title = ""
+
+        if #available(iOS 11.0, *) {
+            viewController.navigationItem.largeTitleDisplayMode = .never
```
Comment (Iamsamwhoami) [inline Hedvig/Screens/ProfileScreen/Screens/MyPayment/MyPayment.swift]: You can now set this on the presentationOptions instead!

Comment (Iamsamwhoami) [inline Hedvig/Screens/ProfileScreen/Screens/MyPayment/MyPayment.swift]: You can now set this on the presentationOptions instead!

File: Hedvig/Screens/ProfileScreen/Screens/MyPayment/MyPayment.swift [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        }
+
+        let form = FormView()
+
+        bag += viewController.install(form) { scrollView in
+            bag += scrollView.chainAllControlResponders(shouldLoop: false, returnKey: .next)
```
Comment (Iamsamwhoami) [inline Hedvig/Screens/ProfileScreen/Screens/MyPayment/MyPayment.swift]: We can remove this I think, we will never have any inputs on this screen so it's not needed.

File: Hedvig/Screens/ProfileScreen/Viewables/MyPaymentRow.swift [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    let presentingViewController: UIViewController
+}
+
+extension MyPaymentRow: Viewable {
+    func materialize(events: SelectableViewableEvents) -> (IconRow, Disposable) {
+        let bag = DisposeBag()
+
+        let row = IconRow(
+            title: String.translation(.PROFILE_PAYMENT_ROW_HEADER),
+            subtitle: "\(monthlyCost) kr/XYZ • \(String.translation(.PROFILE_MY_PAYMENT_METHOD))",
```
Comment (Iamsamwhoami) [inline Hedvig/Screens/ProfileScreen/Viewables/MyPaymentRow.swift]: we should fix support for replacement translations 🤔

File: Hedvig/Screens/ProfileScreen/Viewables/ProfileSection.swift [hunk 4]
```diff
@@ -36,6 +36,15 @@ extension ProfileSection: Viewable {
 
         bag += section.append(myCharityRow)
 
+        let myPaymentRow = MyPaymentRow(
+            monthlyCost: data?.insurance.monthlyCost ?? Int(0),
```
Comment (Iamsamwhoami) [inline Hedvig/Screens/ProfileScreen/Viewables/ProfileSection.swift]: doesn't 0 just work? 🤔

File: Hedvig/Screens/ProfileScreen/Viewables/MyPaymentRow.swift (hunk 3 above)
Comment (isaacSennerholt) [reply Hedvig/Screens/ProfileScreen/Viewables/MyPaymentRow.swift]: @Iamsamwhoami I agree, we should fix it. Though i'm unsure how to do so in this project :/

File: Hedvig/Screens/ProfileScreen/Viewables/ProfileSection.swift (hunk 4 above)
Comment (isaacSennerholt) [reply Hedvig/Screens/ProfileScreen/Viewables/ProfileSection.swift]: @Iamsamwhoami It probably does lol

File: Hedvig/Assets/HedvigColors.swift [hunk 5]
```diff
@@ -18,13 +18,13 @@ extension UIColor {
...
-    static let darkPurple = UIColor(red: 0.03, green: 0.02, blue: 0.27, alpha: 1.0)
-    static let blackPurple = UIColor(red: 0.06, green: 0.00, blue: 0.48, alpha: 1.0)
+    static let blackPurple = UIColor(red: 0.03, green: 0.02, blue: 0.27, alpha: 1.0)
+    static let darkPurple = UIColor(red: 0.06, green: 0.00, blue: 0.48, alpha: 1.0)
     static let darkGray = UIColor(red: 0.61, green: 0.61, blue: 0.67, alpha: 1.0)
     static let lightGray = UIColor(red: 0.91, green: 0.93, blue: 0.94, alpha: 1.0)
     static let offBlack = UIColor(red: 0.25, green: 0.25, blue: 0.31, alpha: 1.0)
     static let offWhite = UIColor(red: 0.98, green: 0.98, blue: 0.99, alpha: 1.0)
-    static let green = UIColor(red: 0.11, green: 0.91, blue: 0.71, alpha: 1.0)
+    static let green = UIColor(red: 0, green: 0.57, blue: 0.46, alpha: 1.0)
```
Comment (Iamsamwhoami) [inline Hedvig/Assets/HedvigColors.swift]: was this not correct before? 🤔

Comment (isaacSennerholt) [reply Hedvig/Assets/HedvigColors.swift]: @Iamsamwhoami Green was the same value as turquoise. Changed it due to a new green color specified in the design. It's quite dark though (#009175), so maybe it should be named darkGreen instead?

File: Hedvig/Generic/Viewables/CircleLabelSmall.swift [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        titleLabel.lineBreakMode = .byWordWrapping
+        titleLabel.textAlignment = .center
+        titleLabel.numberOfLines = 0
+        titleLabel.adjustsFontSizeToFitWidth = true
+        bag += titleLabel.setDynamicText(labelText)
+
+        circleView.layer.shadowOpacity = 0.2
+        circleView.layer.shadowOffset = CGSize(width: 10, height: 10)
+        circleView.layer.shadowRadius = 16
+        circleView.layer.shadowColor = UIColor.darkGray.cgColor
```
Comment (Iamsamwhoami) [inline Hedvig/Generic/Viewables/CircleLabelSmall.swift]: You should always specify a shadow path due to performance savings.

This should work:
```swift
circleView.shadowPath = UIBezierPath(roundedRect: bubbleView.bounds, cornerRadius: circleView.frame.height * 0.5)
```

Add it to the `didLayoutSignal.onValue` on L62

File: Hedvig/Generic/Viewables/CircleLabelWithSubLabel.swift [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+
+        let subLabel = UILabel()
+        subLabel.font = HedvigFonts.circularStdBook?.withSize(18)
+        subLabel.lineBreakMode = .byWordWrapping
+        subLabel.numberOfLines = 0
+        subLabel.textAlignment = .center
+        subLabel.adjustsFontSizeToFitWidth = true
+        bag += subLabel.setDynamicText(subLabelText)
+
+        circleView.layer.shadowOpacity = 0.2
```
Comment (Iamsamwhoami) [inline Hedvig/Generic/Viewables/CircleLabelWithSubLabel.swift]: hmm, it's not that nice to have the same shadow specification scattered around, would be nice if we could keep them in one place 🤔

File: Hedvig/Screens/ProfileScreen/Screens/MyPayment/MyPayment.swift [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+
+        let viewController = UIViewController()
+        viewController.title = String.translation(.MY_PAYMENT_TITLE)
+
+        let form = FormView()
+        bag += viewController.install(form)
+
+        bag += client.fetch(query: ProfileQuery()).onValue { result in
+            if let insurance = result.data?.insurance {
+                let monthlyPaymentCircle = MonthlyPaymentCircle(monthlyCost: insurance.monthlyCost ?? 0)
```
Comment (Iamsamwhoami) [inline Hedvig/Screens/ProfileScreen/Screens/MyPayment/MyPayment.swift]: This line is quite long, I would say line break after the `(`.

File: Hedvig/Generic/Viewables/CircleLabelWithSubLabel.swift (hunk 7 above)
Comment (isaacSennerholt) [reply Hedvig/Generic/Viewables/CircleLabelWithSubLabel.swift]: @Iamsamwhoami I agree. Will fix 👍

File: Hedvig/Screens/ProfileScreen/Screens/MyPayment/MyPayment.swift (hunk 8 above)
Comment (Iamsamwhoami) [reply Hedvig/Screens/ProfileScreen/Screens/MyPayment/MyPayment.swift]: Would be nice if `swiftformat` did this for us but I am afraid it's not capable of that 🙃

File: Hedvig/Assets/HedvigColors.swift (hunk 5 above)
Comment (Iamsamwhoami) [reply Hedvig/Assets/HedvigColors.swift]: @isaacSennerholt https://github.com/HedvigInsurance/brand/blob/master/colors.ts#L14 green is `0.11, 091, 0.71` 🤔 ask Zak about what we should call this new one.

Comment (isaacSennerholt) [reply Hedvig/Assets/HedvigColors.swift]: @Iamsamwhoami But that doesn't make sense though, as the green you're referring to in brand repo also is same value as turquoise. Why is green and turquoise the same to begin with? Can this new color, which is more green-like than turquoise, be green for now? If not, we ought to review the old green color and try to change it's use to turquoise.

Comment (Iamsamwhoami) [reply Hedvig/Assets/HedvigColors.swift]: @isaacSennerholt it doesn't make sense, but turquoise isn't really used (only once in the old app) and "#1BE9B6" isn't turquoise 😅it's green. So I guess this new one should be called darkGreen?  🤷‍♂️

Comment (Iamsamwhoami) [reply Hedvig/Assets/HedvigColors.swift]: @isaacSennerholt Hmm, I would say remove green, call this new one darkGreen.

**Pull request merged**